parser.go

```diff
@@ -1072,7 +1124,16 @@ func (p *parser) getSchemaObjectCached(pkgPath, pkgName, typeName string) (*Sche
 	var schemaObject *SchemaObject
 
 	// see if we've already parsed this type
-	if knownObj, ok := p.KnownIDSchema[genSchemeaObjectID(pkgName, typeName)]; ok {
+	knownObj := checkCache(pkgName, typeName, p)
```

Comment:
Instead of passing `p` into everything, I think the pattern is to make them functions that operate on `p` with

```
function (p *parser) checkCache (pkgName string, typeName string) {
```

executed as

```
p.checkCache(pkgName, typeName)
```

Though admittedly I am new to go and this pattern.